src/read.c

```diff
@@ -3337,9 +3357,72 @@ avifResult avifDecoderParse(avifDecoder * decoder)
     return avifDecoderReset(decoder);
 }
 
-static avifCodec * avifCodecCreateInternal(avifCodecChoice choice)
+static avifCodec * avifCodecCreateInternal(avifCodecChoice choice, avifTile * tile, avifDiagnostics * diag)
```

Comment:
Declare `tile` as const. Alternatively, pass `operatingPoint` and `allLayers` instead.

Reply:
Done.

I chose to pass `tile` instead of specific parameters since some of the callers pass stuff like `&decoder->data->tile.tile[1]`. It would mean converting this into a local variable in the caller since it's too cumbersome to repeat this in the caller.